cc/addmultisig_request.go

```diff
@@ -0,0 +1,169 @@
+package cc
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/anoideaopen/acl/helpers"
+	"github.com/btcsuite/btcutil/base58"
+	"github.com/hyperledger/fabric-chaincode-go/shim"
+)
+
+const (
+	base58Signatures = true
+	hexSignatures    = false
+)
+
+func indexOf(arg string, order []string) int {
+	for i, name := range order {
+		if name == arg {
+			return i
+		}
+	}
+	return indexNotFound
+}
+
+const indexNotFound = -1
+
+const (
+	argRequestID         = "requestID"
+	argChaincodeID       = "chaincodeID"
+	argChannelID         = "channelID"
+	argKeysRequired      = "keysRequired"
+	argNonce             = "nonce"
+	argKeysAndSignatures = "keysAndSignatures"
+)
+
+type AddMultisigRequest struct {
+	RequestID               string
+	ChaincodeName           string
+	ChannelName             string
+	PublicKeys              []PublicKey
+	RequiredSignaturesCount int
+	Signatures              [][]byte
+	Message                 string
+	SignedTx                []string
+	Nonce                   string
+}
+
+func (request *AddMultisigRequest) parseArguments(
```

Comment:
Could u rename the receiver like 'r' or 'req' or smth else? Coz it's not ideomatic now.

Reply:
Done